ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -621,6 +621,80 @@ export class AptosClient {
     const txnHash = await this.generateSignSubmitTransaction(sender, payload, extraArgs);
     return this.waitForTransactionWithResult(txnHash, extraArgs);
   }
+
+  /**
+   * Rotate an account's auth key. After rotation, only the new private key can be used to sign txns for
```

Comment:
after rotation, we also need to update the `forAccount`'s private and public key to new private and public key.

Reply:
Unfortunately, the key pair of `AptosAccount` is not mutable as of now. Wallet and other dApps will have to update the keys in their persistent storage and create a new `AptosAccount`.

Reply:
will they be able to create a new `AptosAccount` with the same address?

Reply:
yes. They can specify a private key and address as parameters when creating an `AptosAccount`

Reply:
Can we add a comment to this function at least like:
```
WARNING: You must create a new instance of AptosAccount after using this function.
```

Reply:
I'd like to better understand this flow. It doesn't seem like our current account model cleanly applies for wallets at all. That they would likely need to create their own version of this and then do what they need to do. This seems like an abnormal paradigm.

Reply:
It should work well with wallets. Wallets do the key management already. SDK just provides them means to rotate the key. Upon successful rotation, wallets need to save the new key in their local storage and create an AptosAccount instance and discard the old AptosAccount instance.

Reply:
Just brainstorming. What if `AptosAccount` was only `{  privateKey, publicKey }`?

Then we can have an async function on `AptosAccount` for the address.

Reply:
@jjleng Another idea: Should we put this function under `AptosAccount` rather than passing an account to the client?

Reply:
Logically, yes. I actually tried. However, that would make `AptosAccount` depend on `AptosClient`, which creates a circular dependency.

Reply:
yea would probably be good is AptosClient didn't depend on AptosAccount